### PR TITLE
feat: venom as Lib

### PR DESCRIPTION
**Pull request by fsamin**
Sample usage

```golang
func TestVenomTestCase(t *testing.T) {
	v := venom.TestCase(t, "TestVenomTestCase", venom.V{
		"foo": "bar",
	})
	r := v.Do(venom.H{
		"type":   "exec",
		"script": "echo {{.foo}}",
	})
	assert.Equal(t, "bar", r["result.systemout"])
}
```

still have to improve it; with some docs also

### CODE REVIEW

**Comment by yesnault** [comment]
close #23

**Comment by yesnault** [comment]
close #5

File: executors/imap/decode.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 	"net/mail"
 
 	log "github.com/Sirupsen/logrus"
+	"github.com/runabove/venom"
```
Comment (sguiheux) [inline executors/imap/decode.go]: wrong place

File: lib/lib.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+			Name:      name,
+		},
+		&venom.TestCase{
+			Name: name,
+		},
+		name,
+	}
+}
+
+//Do execuutes a veom test steps
```
Comment (sguiheux) [inline lib/lib.go]: only one u

File: lib/lib.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	venom.RegisterExecutor(readfile.Name, readfile.New())
+	venom.RegisterExecutor(smtp.Name, smtp.New())
+	venom.RegisterExecutor(ssh.Name, ssh.New())
+	venom.RegisterExecutor(web.Name, web.New())
+	venom.RegisterTestCaseContext(defaultctx.Name, defaultctx.New())
+	venom.RegisterTestCaseContext(webctx.Name, webctx.New())
+}
+
+//H is a map of test parameters
+type H map[string]interface{}
```
Comment (sguiheux) [inline lib/lib.go]: P

File: lib/lib.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		},
+		&venom.TestCase{
+			Name: name,
+		},
+		name,
+	}
+}
+
+//Do execuutes a veom test steps
+func (t *T) Do(teststep H) R {
```
Comment (sguiheux) [inline lib/lib.go]: testStepParam

**Pull request merged**